test/exercism/user_contribution_test.rb

```diff
@@ -0,0 +1,37 @@
+require_relative '../test_helper'
+require_relative '../x_helper'
+require_relative '../integration_helper'
+
+class UserContributionTest < Minitest::Test
+  include DBCleaner
+
+  def setup
+    super
+    @user = User.create
+    @f = './test/fixtures/xapi_v3_tracks.json'
+    create_exercise('animal', 'dog')
+    create_exercise('fake', 'hello-world')
+    create_exercise('fake', 'two')
+  end
+
+  def test_user_contributions
+    X::Xapi.stub(:get, [200, File.read(@f)]) do
+      actual = UserContribution.new(@user).list
+      expected = {
+        completed_tracks: [
+          {
+            'completed_exercise' => "1", 'language' => 'animal',
+            'total_in_track' => 1, 'language_name' => 'Animal'
+          },
+        ],
+      }
+      assert_equal expected, actual
+    end
```

Comment:
Let's simplify the names here:

- `tracks` (instead of `completed_tracks`)
- `id` (the track id)
- `language` (the language name)
- `problem_count`
- `solution_count`